pandas/core/frame.py

```diff
@@ -387,8 +387,12 @@ def _constructor_expanddim(self):
     # Constructors
 
     def __init__(self, data=None, index=None, columns=None, dtype=None, copy=False):
+        if data is None and index is not None and columns is not None and dtype is None:
+            data = np.nan
+
         if data is None:
```

Comment:
I think this will always override the previous code block

Reply:
No. 
Because when the if in line 390 is true, data becomes np.nan
and the following if evaluates to  False because np.nan is not None

```python
>>> import numpy
>>> numpy.nan is None
False
```

Reply:
Sorry, you're right